client/src/model/values/instantaneousValues/InstantaneousValueI.hx

```diff
@@ -0,0 +1,49 @@
+package model.values.instantaneousValues;
+
+/**
+ * @author AdvaitTrivedi
+ */
+interface InstantaneousValueI 
+{
+	/**
+	 * Returns a string representation of the signal value.
+	 * @return
+	 */
+	public function toString() : String;
+	
+	/**
+	 * performs Logical AND operation on the passed signalValue parameter with the this value. 
+	 * @param	signalValue
+	 * @return
+	 */
+	@:allow(model.gates.AND)
```

Comment:
We don't need these operations yet.

Reply:
They will be used at some point in time, I have them implemented already... so.... ?

Reply:
You can leave them if you want.  I didn't really look them over too carefully.  I have yet to document how they should behave.